media-proxy/src/mesh/st2110_22tx.cc

```diff
@@ -0,0 +1,88 @@
+#include "st2110tx.h"
+
+namespace mesh::connection {
+
+ST2110_22Tx::ST2110_22Tx() {}
+
+ST2110_22Tx::~ST2110_22Tx() {}
+
+st_frame *ST2110_22Tx::get_frame(st22p_tx_handle h) { return st22p_tx_get_frame(h); };
+
+int ST2110_22Tx::put_frame(st22p_tx_handle h, st_frame *f) { return st22p_tx_put_frame(h, f); };
+
+st22p_tx_handle ST2110_22Tx::create_session(mtl_handle h, st22p_tx_ops *o)
+{
+    return st22p_tx_create(h, o);
+};
+
+int ST2110_22Tx::close_session(st22p_tx_handle h) { return st22p_tx_free(h); };
+
+Result ST2110_22Tx::configure(context::Context &ctx, const std::string &dev_port,
+                              const MeshConfig_ST2110 &cfg_st2110,
+                              const MeshConfig_Video &cfg_video)
+{
+    static int session_id;
+    static std::mutex mtx;
+    std::lock_guard<std::mutex> lock(mtx);
+
+    if (cfg_st2110.transport != MESH_CONN_TRANSPORT_ST2110_22) {
+        set_state(ctx, State::not_configured);
+        return set_result(Result::error_bad_argument);
+    }
+
+    _st = get_mtl_handle(dev_port, MTL_LOG_LEVEL_CRIT, cfg_st2110.local_ip_addr);
+    if (_st == nullptr) {
+        log::error("Failed to get MTL device");
+        set_state(ctx, State::not_configured);
+        return set_result(Result::error_bad_argument);
+    }
+
+    char session_name[NAME_MAX] = "";
+
+    snprintf(session_name, NAME_MAX, "mcm_tx_st22_%d", session_id++);
```

Comment:
`session_id` is not thread safe. Be careful!

Reply:
Fixed during configure() rework https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/261/commits/d6fb54f34d3b43543423c71401f5fca75f1fbdde